### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by MSpake**

### CODE REVIEW

File: Controllers/UsersController.cs [hunk 1]
```diff
@@ -1,44 +1,113 @@
...
+      if (user == null)
+      {
+        return NotFound();
+      }
+      if (user != null)
+      {
+        user.Password = null;
+      }
+
+      var currentUserId = int.Parse(User.Identity.Name);
```
Comment (vladimirsan) [inline Controllers/UsersController.cs]: Is there a way to avoid var here?

File: Controllers/UsersController.cs [hunk 2]
```diff
@@ -1,44 +1,113 @@
...
+      {
+        Subject = new ClaimsIdentity(new Claim[]
+          {
+            new Claim(ClaimTypes.Name, user.Id.ToString()),
+            new Claim(ClaimTypes.Role, user.Role)
+          }),
+        Expires = DateTime.UtcNow.AddDays(7),
+        SigningCredentials = new SigningCredentials(new SymmetricSecurityKey(key), SecurityAlgorithms.HmacSha256Signature)
+      };
+      var token = tokenHandler.CreateToken(tokenDescriptor);
```
Comment (vladimirsan) [inline Controllers/UsersController.cs]: same here with var

File: Controllers/UsersController.cs [hunk 3]
```diff
@@ -1,44 +1,113 @@
...
+      user.Token = tokenHandler.WriteToken(token);
+
+      // remove password before returning
```
Comment (vladimirsan) [inline Controllers/UsersController.cs]: I don't think this comment is neccesary

File: Controllers/UsersController.cs [hunk 4]
```diff
@@ -1,44 +1,113 @@
...
+      string email = userFromFrontend.Email;
+      string password = userFromFrontend.Password;
+      var user = _context.Users.FirstOrDefault(x => x.Email == email && x.Password == password);
+
+      if (user == null)
+      {
+        return NotFound();
+      }
+      JwtSecurityTokenHandler tokenHandler = new JwtSecurityTokenHandler();
+      var key = Encoding.ASCII.GetBytes(Configuration["Secret"]);
```
Comment (vladimirsan) [inline Controllers/UsersController.cs]: var

File: Controllers/UsersController.cs [hunk 5]
```diff
@@ -1,44 +1,113 @@
...
+      var tokenDescriptor = new SecurityTokenDescriptor
```
Comment (vladimirsan) [inline Controllers/UsersController.cs]: var

**Review by vladimirsan — CHANGES_REQUESTED** [review]

File: Controllers/UsersController.cs (hunk 1 above)
Comment (MSpake) [reply Controllers/UsersController.cs]: Done

File: Controllers/UsersController.cs (hunk 3 above)
Comment (MSpake) [reply Controllers/UsersController.cs]: Removed

File: Controllers/UsersController.cs (hunk 4 above)
Comment (MSpake) [reply Controllers/UsersController.cs]: Done

File: Controllers/UsersController.cs (hunk 2 above)
Comment (MSpake) [reply Controllers/UsersController.cs]: Done

File: Controllers/UsersController.cs (hunk 5 above)
Comment (MSpake) [reply Controllers/UsersController.cs]: Done

**Review by vladimirsan — APPROVED** [review]

**Pull request merged**